### ISSUE TITLE
MultiScale Training

### ISSUE THREAD

**Issue by karanchahal** [experiencor/keras-yolo2#45]
Hey , I want to incorporate MultiScale training into your model. Any idea what resources/papers I should read and implement this.

**Comment by experiencor** [comment]
@karanchahal It would be great if you can implement it. You may refer to https://arxiv.org/abs/1612.08242 for details. The basic idea is to change the size of the input image now and then. My code works with any fixed input size, but not changing input size. You will have to change the way that I handle the cell_grid variable in custom_loss function (in frontend.py).

**Comment by karanchahal** [comment]
Hey I was reading through your code.
I encountered a line 
self.true_boxes = Input(shape=(1, 1, 1, max_box_per_image , 4))  
in frontend.py

What does true_boxes intend to do and why does it have 5 dimensions ?

**Comment by experiencor** [comment]
@karanchahal self.true_boxes contains the list of all bounding boxes labels, meaning (x, y, w, h)'s. The first three dimensions are actually redundant but help to simply Tensorflow code in the custom loss function. You refer to https://github.com/experiencor/basic-yolo-keras/issues/42 for further details.

**Comment by Steven-N-Hart** [comment]
Sorry, I still don't understand the shape parameters. Could you provide an example? I understand the `max_box_per_image`.

**Comment by experiencor** [comment]
@Steven-N-Hart It has the first 3 redundant dimensions jus out of the convenience for the computation of the loss function. The usage of true_boxes starts from line 155 in frontend.py.

**Comment by experiencor** [comment]
Multiscale training is supported here (https://github.com/experiencor/keras-yolo3).
